client/my-sites/checkout/composite-checkout/hooks/use-remove-from-cart-and-redirect.ts

```diff
@@ -25,8 +27,28 @@ export default function useRemoveFromCartAndRedirect(
 	removeProductFromCartAndMaybeRedirect: RemoveProductFromCart;
 } {
 	const { removeProductFromCart } = useShoppingCart();
-	// The cloud.jetpack.com/pricing page sends a `checkoutBackUrl` url query param to checkout.
-	const { checkoutBackUrl } = useSelector( getInitialQueryArguments ) as { [ k: string ]: string };
+
+	// In some cases, the cloud.jetpack.com/pricing page sends a `checkoutBackUrl` url query param to checkout.
+	const { checkoutBackUrl } = useSelector( getInitialQueryArguments ) ?? {};
+
+	// Verify the checkBackUrl query arg is Jetpack Cloud /pricing page or the site itself.
+	const isCheckoutBackUrlValid = useMemo( () => {
+		if ( ! checkoutBackUrl ) {
+			return false;
+		}
+		const theSite = siteSlug || siteSlugLoggedOutCart;
+		const allowedHosts = [ 'jetpack.cloud.localhost', 'cloud.jetpack.com', theSite ];
+		const { hostname } = parseUrl( checkoutBackUrl, true, true );
```

Comment:
I think you can use `const { hostname } = new URL( checkoutBackUrl )` instead of needing to import (the deprecated) `parseUrl`, although you may want to wrap it in a `try/catch` just in case the URL isn't valid.